From: Jiri Slaby <jslaby@suse.cz>
Subject: kABI: protect struct se_lun
Patch-mainline: never, kabi
References: kabi

In 4.4.81, commit 2bf7791c6a5ec21228383f5a1fb5c443c3336622 (target:
Avoid mappedlun symlink creation during lun shutdown), upstream commit
49cb77e297dc611a1b795cfeb79452b3002bd331 added a bool to struct se_lun.
This indeed changed the layout of the structure and the kABI checker now
complains.

Given this is only a boolean and it is next to another bool in SP2
(after
patches.drivers/0034-target-remove-enum-transport_lunflags_table), we
use the hole and just make the changes invisible to the kABI checker by
the usual #ifndef.

Signed-off-by: Jiri Slaby <jslaby@suse.cz>
---
 include/target/target_core_base.h |    2 ++
 1 file changed, 2 insertions(+)

--- a/include/target/target_core_base.h
+++ b/include/target/target_core_base.h
@@ -716,7 +716,9 @@ struct se_lun {
 #define SE_LUN_LINK_MAGIC			0xffff7771
 	u32			lun_link_magic;
 	u32			lun_access;
+#ifndef __GENKSYMS__
 	bool			lun_shutdown;
+#endif
 	u32			lun_index;
 
 	/* RELATIVE TARGET PORT IDENTIFER */
